### PR TITLE
tests added for enableSemanticdb extension property

### PR DESCRIPTION
**Pull request by fsanjuan**
Funcional test class added to verify enableSemanticdb adds compileScala task dependency on scala fix tasks

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -7,6 +8,17 @@ plugins {
...
+gradlePlugin {
+    plugins {
+        scalafix {
+            id = 'io.github.cosmicsilence.scalafix'
+            implementationClass = 'io.github.cosmicsilence.scalafix.ScalafixPlugin'
+        }
+    }
+}
+
+version '0.0.1-SNAPSHOT'
```
Comment (marcelocenerine) [inline build.gradle]: why do we need the changes above?

File: src/test/groovy/io/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        buildFile.write'''
+plugins {
+    id 'scala'
+    id 'io.github.cosmicsilence.scalafix'
+}
+
+repositories {
+    mavenCentral()
+}
+'''
```
Comment (marcelocenerine) [inline src/test/groovy/io/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy]: would this indentation look anybetter with `.stripMargin()`?

**Comment by marcelocenerine** [comment]
The tests look great, @fsanjuan 👍 . I just left two questions for you

File: build.gradle (hunk 1 above)
Comment (fsanjuan) [reply build.gradle]: Apologies, that was a merge conflict I didn't solve properly :)

File: src/test/groovy/io/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy (hunk 2 above)
Comment (marcelocenerine) [reply src/test/groovy/io/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy]: ignore. Looks like all the examples I found on the official docs don't do that

File: build.gradle (hunk 1 above)
Comment (marcelocenerine) [reply build.gradle]: regarding the overlapping configuration of the plugin: I did some research on this and found this: https://docs.gradle.org/5.6.2/userguide/test_kit.html#sub:test-kit-automatic-classpath-injection . After playing around with the properties file and the snippet you added, it looks like having only the properties file is enough provided that you run tests using gradle as runner. If you attempt to run the test through Intellij then they will fail.

I also tested with only the config in build.gradle (deleted the properties file(. Looks like that is enough to get the plugin working. I tested it on my scala test project and it worked. I'd vote for keeping only that approach and avoid the redundant config.

**Pull request merged**